### ISSUE TITLE
Please use https method over ssh

### ISSUE THREAD

**Issue by frederic-loui** [NetworkGradeLinux/mion#66]
Hi, (Currently going through/testing Mion documentation "[Getting started](https://docs.mion.io/getting-started.html)": )

SSH method yield an error for random user (like me), please refer to `https`  method instead.

```
git clone --recursive git@github.com:NetworkGradeLinux/mion.git
Cloning into 'mion'...
The authenticity of host 'github.com (140.82.121.4)' can't be established.
RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Warning: Permanently added 'github.com,140.82.121.4' (RSA) to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```
with `https`:
```
git clone https://github.com/NetworkGradeLinux/mion.git
Cloning into 'mion'...
remote: Enumerating objects: 131, done.
remote: Counting objects: 100% (131/131), done.
remote: Compressing objects: 100% (64/64), done.
remote: Total 853 (delta 70), reused 92 (delta 58), pack-reused 722
Receiving objects: 100% (853/853), 211.23 KiB | 1.04 MiB/s, done.
Resolving deltas: 100% (401/401), done.

```
PS: Labels: Documentation
      Same remarks for `meta-mion` and `meta-mion-bsp`

A bientôt,
Frederic

**Comment by CommitThis** [comment]
Hi Frederic, have you added your public key to your GitHub account?

**Comment by frederic-loui** [comment]
Hi, 
No, but I've added it now upon your feedback. But IMHO not all users put their SSH public key.
Let me try again then.

**Comment by CommitThis** [comment]
As an aside, I'm not sure it is possible to change the default method in the drop down box provided on the repository page, presuming that was what you were originally referring to. I looked at the settings for all of my personal repositories and a cursory glance revealed no option for doing so. However, I am not an administrator for this project so I have no way of checking if there is anything different there.
If we were to make the change, the `SRC_URI`s for the repositories that the project is trying to pull in would have to have their protocol changed too; there may be good reasons why SSH is preferred over HTTPS in that case.

**Comment by frederic-loui** [comment]
Hi K. !

I confirmed that:

0- (Having a GitHub account beforehand)
1- Putting my SSH pub key on GitHub
2- Upload my SSH private key on my test server
3- Activate SSH-Agent
4- Add my key to SSH-Agent

Solved the issue !

But, I do feel that this is not a straightforward way to introduce Mion to new users, especially to user not familiar to all the steps above ... I mean this can prevent adoption ... It is up to your decision to keep `ssh` (I'd rather use `https`) but it does force users to have done beforehand all of the above steps... Also please note that that this force people to have a GitHub account.

If `ssh` method is kept, this needs to be mentioned in the `Pre-requirements` section.

Last but not least, in `README.md` the transport method suggested in `https` while in the documentation it is `ssh`. Of course this does not align with `SRC_URI` used in `../mc_build.sh `.

Build has been launched now ! Let's wait few hours ... :)

PS: This issue can be closed.

**Comment by CommitThis** [comment]
I'll leave the ticket open for now, so that @toganlabs is able to comment on the implications of changing the method. However, I take your point on the README, and will raise a PR at some point for changing it to `ssh` for the time being.

By the way, with respect to step 2, you could instead run `ssh-keygen` on your development host. Steps 3 and 4 may be unnecessary -- `ssh` without the agent may prompt you to unlock the key if it has a password, and then connect.

**Comment by frederic-loui** [comment]
Without ssh-agent mc_build.sh consistently fails.

Le sam. 12 déc. 2020 à 16:43, CommitThis! <notifications@github.com> a
écrit :

**Comment by CommitThis** [comment]
Fair enough.

**Comment by CommitThis** [comment]
So `git clone --recursive https://github.com/NetworkGradeLinux/mion.git` should work without any changes.

The fetcher issues where my fault. I specified SSH in the `SRC_URI`

@frederic-loui would you be able to quickly review by cloning Mion, and when it gets to manually cloning `meta-mion-bsp`  could you checkout `CommitThis/src-uri-not-ssh`?

PR in https://github.com/NetworkGradeLinux/meta-mion-bsp/pull/38

**Comment by CommitThis** [comment]
Back to to-do, due to documentation changes.

**Comment by frederic-loui** [comment]
Hi ! I used  [`getting started`](https://docs.mion.io/getting-started.html)  and also used `meta-mion-bsp`/`CommitThis/src-uri-not-ssh`. Build started (and is still running ...) You should use `https` in documentation so that you are consistent with all other docs docs. Like this `[README.md](https://github.com/NetworkGradeLinux/mion/blob/dunfell/README.md)` 
Otherwise it seems to be prefect !

**Comment by kprosise** [comment]
Recent commit should change this. In general we are trying to withhold making changes to documentation until versioning is implemented, but an exception was made in this case.
